src/run_program.rs

```diff
@@ -244,8 +244,17 @@ where
         };
 
         let op_atom = match self.allocator.sexp(&op_node) {
-            SExp::Pair(_, _) => {
-                return Node::new(self.allocator, program.clone()).err("(()) eval disabled")
+            SExp::Pair(new_operator, must_be_nil) => {
+                if let SExp::Atom(_) = self.allocator.sexp(&new_operator) {
```

Comment:
Is there an implicit check for is_listp(new_operator) that I'm not seeing?

Reply:
Yeah, the `if let SExp::Atom(_) =` only succeeds if `new_operator` represents an atom.